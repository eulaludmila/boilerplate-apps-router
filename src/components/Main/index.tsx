import * as S from './styles'

const Main = () => {
  return (
    <S.Wrapper>
      <h1>Hello, world!</h1>
    </S.Wrapper>
  )
}

export default Main
