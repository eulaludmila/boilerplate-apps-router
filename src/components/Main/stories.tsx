import { Meta, StoryObj } from '@storybook/react'
import Main from '.'

export default {
  title: 'Main',
  component: Main,
  parameters: {
    layout: 'fullscreen'
  }
} as Meta //funciona para configurar um story de um componente

export const Default: StoryObj = {}
