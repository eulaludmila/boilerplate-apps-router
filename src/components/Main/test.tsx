import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the reading', () => {
    //renderiza o componente
    const { container } = render(<Main />)
    //busca o elemento e verifica a existÃªncia dele
    expect(
      screen.getByRole('heading', { name: /hello, world!/i })
    ).toBeInTheDocument()

    //gerar snapshot
    expect(container.firstChild).toMatchSnapshot()
  })
})
