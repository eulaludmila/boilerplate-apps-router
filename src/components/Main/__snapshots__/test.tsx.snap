// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the reading 1`] = `
<main
  class="sc-braxZu hCvKmL"
>
  <h1>
    Hello, world!
  </h1>
</main>
`;
